### PR TITLE
add examples for keyPhrase and percentage

### LINKED ISSUE #2904: Percentage template should be a number
Percentage is currently modeled as a string type.  It should be a numeric type with its own help templates.

### PR DESCRIPTION
**Pull request by coldplaying42**
Fixes #2904

## Proposed Changes

  - 
  -
  -

### CODE REVIEW

File: experimental/generation/generator/packages/library/templates/standard/percentage-add-string.dialog.lg [hunk 1]
```diff
@@ -0,0 +1,8 @@
+[generator.lg](../generator.lg)
+
+# filename
+- ${addFilename('number')}
```
Comment (chrimc62) [inline experimental/generation/generator/packages/library/templates/standard/percentage-add-string.dialog.lg:4]: I don't think we need this--percentage should always be a number.

File: experimental/generation/generator/packages/library/templates/standard/percentage-remove-string.dialog.lg [hunk 2]
```diff
@@ -0,0 +1,8 @@
+[generator.lg](../generator.lg)
+
+# filename
+- ${removeFilename('number')}
```
Comment (chrimc62) [inline experimental/generation/generator/packages/library/templates/standard/percentage-remove-string.dialog.lg:4]: I don't think we need this file.

File: experimental/generation/generator/packages/library/templates/standard/percentage.schema [hunk 3]
```diff
@@ -1,14 +0,0 @@
...
-    "$schema": "http://json-schema.org/draft-07/schema",
-    "type": "string",
-    "$entities": [
-        "percentage"
-    ],
-    "$templates": [
-        "string",
-        "percentageEntity.lu",
-        "percentageEntity.lg",
-        "percentageEntity-add-string.dialog",
```
Comment (chrimc62) [inline experimental/generation/generator/packages/library/templates/standard/percentage.schema:11]: Why did you delete this file?  We need this so that someone can $ref a percentage.  The change was to make the type a number.

File: experimental/generation/generator/packages/library/templates/standard/percentageEntity-remove-string.dialog.lg [hunk 4]
```diff
@@ -1,8 +1,8 @@
 [generator.lg](../generator.lg)
 
 # filename
-- ${removeFilename('percentage')}
+- ${removeFilename('number')}
 
 # template
```
Comment (chrimc62) [inline experimental/generation/generator/packages/library/templates/standard/percentageEntity-remove-string.dialog.lg:6]: >template [](start = 2, length = 8)

We don't need this file.

File: experimental/generation/generator/packages/library/templates/standard/en-us/percentageEntity.en-us.lu.lg [hunk 5]
```diff
@@ -1,9 +1,11 @@
 [standard.en-us.lg](standard.en-us.lg)
 
 # filename
-- ${builtinEntityFilename('percentage')}
+- ${builtinEntityFilename('number')}
 
 # template
 - ```
-@ prebuilt percentage
+@ prebuilt number
```
Comment (chrimc62) [inline experimental/generation/generator/packages/library/templates/standard/en-us/percentageEntity.en-us.lu.lg:8]: >number [](start = 11, length = 6)

This is incorrect.  We should use the percentage prebuilt.  The resolution is a number and the json schema type should be number.

File: experimental/generation/generator/packages/library/templates/standard/en-us/percentageEntity.en-us.lu.lg [hunk 6]
```diff
@@ -1,9 +1,11 @@
...
+
+${operations('number', [1, 20.5, 51, 99.8], false)}
```
Comment (chrimc62) [inline experimental/generation/generator/packages/library/templates/standard/en-us/percentageEntity.en-us.lu.lg:10]: >1, 20.5, 51, 99.8 [](start = 24, length = 17)

There should be strings like "3%", "2.5 percent", etc.

File: experimental/generation/generator/packages/library/templates/standard/en-us/percentageProperty.en-us.lg.lg [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+                        clearDefault())}
+${validationTemplates()}
+${helpTemplates(repromptPercentage(), helpPercentage())}
+${showTemplate(showDefault())}
+${nameTemplate()}
+${valueTemplate()}
+```
+
+# missingPercentage()
+- Enter an percentage for ${propertyName()} ${numberRanges()}
```
Comment (chrimc62) [inline experimental/generation/generator/packages/library/templates/standard/en-us/percentageProperty.en-us.lg.lg:23]: >an [](start = 8, length = 2)

a

File: experimental/generation/generator/packages/library/test/forms/unittests.form [hunk 8]
```diff
@@ -91,10 +91,7 @@
             ]
         },
         "Percentage": {
-            "type": "string",
-            "$entities": [
-                "percentage"
-            ]
+            "type": "percentage"
```
Comment (chrimc62) [inline experimental/generation/generator/packages/library/test/forms/unittests.form:94]: >percentage [](start = 21, length = 10)

This is not a valid JSON schema type.  This is why we had the schema, you should do $ref percentage.schema which has a type of number.

File: experimental/generation/generator/packages/library/test/forms/unittest_percentage.form [hunk 9]
```diff
@@ -3,10 +3,7 @@
     "type": "object",
     "properties": {
         "Percentage": {
-            "type": "string",
-            "$entities": [
-                "percentage"
-            ]
+            "type": "percentage"
```
Comment (chrimc62) [inline experimental/generation/generator/packages/library/test/forms/unittest_percentage.form:6]: >percentage [](start = 21, length = 10)

Use $ref

File: experimental/generation/generator/packages/library/test/forms/unittest_percentage_with_limits.form [hunk 10]
```diff
@@ -3,7 +3,9 @@
     "type": "object",
     "properties": {
         "Percentage": {
-           "$ref": "template:percentage.schema"
+            "type": "percentage",
```
Comment (chrimc62) [inline experimental/generation/generator/packages/library/test/forms/unittest_percentage_with_limits.form:6]: Use $ref

**Comment by chrimc62** [comment]
We should be using percentage.schema which defines the type as number.  Percentage is not a valid JSON schema type.

**Review by chrimc62 — CHANGES_REQUESTED** [review]
:clock1:

**Review by chrimc62 — APPROVED** [review]
:shipit:

**Comment by coldplaying42** [comment]
@boydc2014 @GeoffCoxMSFT @microsoft/bb-samples Please help me approve the PR.
